otpInput/otpInput.js

```diff
@@ -0,0 +1,148 @@
+export class OtpInput {
+  form = document.createElement('form');
+  inputQty = 6;
+  inputs = [];
+  input = document.createElement('input');
+  otp = '';
+  inputMatchesOtp = false;
+
+  constructor(inputQty = 6) {
+    this.form.className = 'otp-form';
+    this.inputQty = inputQty;
+    this.input.type = 'hidden';
+    this.input.name = 'otp';
+  }
+
+  getOtp() {
+    return this.input.value;
+  }
+
+  setOtp(otp) {
+    this.otp = otp;
+  }
+
+  resetOtp() {
+    this.input.value = '';
+    this.otp = '';
+    this.inputs.forEach((input) => {
+      input.value = '';
+    });
+  }
+
+  createElements() {
+    for (let i = 0; i < this.inputQty; i++) {
+      const inputElem = document.createElement('input');
+      inputElem.type = 'number';
+      inputElem.className = 'otp-input';
+      inputElem.maxLength = 1;
+      inputElem.pattern = 'd*';
+      inputElem.id = `otp-input-${i}`;
+      this.form.appendChild(inputElem);
+      this.inputs.push(inputElem);
+    }
+    document.body.appendChild(this.form);
+    this.addEventListeners();
+  }
+
+  addEventListeners() {
+    for (let i = 0; i < this.inputs.length; i++) {
+      const currentInput = this.inputs[i];
+
+      currentInput.addEventListener('input', () => {
+        if (currentInput.value.length == 1 && i + 1 < this.inputs.length) {
+          this.inputs[i + 1].focus();
+        }
+
+        if (currentInput.value.length > 1) {
+          if (isNaN(currentInput.value)) {
+            input.value = '';
+            this.updateInput();
+            return;
+          }
+
+          const chars = currentInput.value.split('');
+
+          for (let pos = 0; pos < chars.length; pos++) {
+            if (pos + i >= this.inputs.length) break;
+
+            let targetInput = this.inputs[pos + i];
+            targetInput.value = chars[pos];
+          }
```

Comment:
I'm a little confused about what the code in this `for` loop is doing/needed for